src/main.cpp

```diff
@@ -547,6 +791,124 @@ int main(int argc, char *argv[]) {
 
  cout << "Saving results... " << endl;
 
+
+#if defined(USEIRAPL)
+
+	 is_log_pwr = false;
+	 is_log_pwr_thread.join();
+
+ if (is_p_rate > 0)
+ {
+	 std::vector<std::string> time_strings;
+
+	 for (size_t i = 0; i < is_pwr_time.size()-1; i++) {
```

Comment:
Why do you skip the last element here and below?

Reply:
Because I start counting at zero? And the range of the vecotr is [0..is_pwr_time.size()-1]. Or did I miss something?

Reply:
Stating at zero is okay if you use `for (size_t i = 0; i < is_pwr_time.size(); i++) {`. The reason seems to be that there are differences `is_pwr0[i].at(j+1)-is_pwr0[i].at(j)` below.

Reply:
Ah of course, as the logged values are energy and not power, I have to take the difference (or derivative) to get the power. Hence, I also had to subtract one from the time to get the same length vector. I did not want to calculate the difference directly in the logging thread to save CPU time.

Reply:
The timestamps are logged accordingly in the thread to account for this circumstance.

Reply:
Isn't energy the more interesting quantity in the end?

Reply:
The problem is that the implementations that measure power directly and do not calculate it heuristically output power as direct measurement and not energy. Therefore, we can easily compare the values. The energy counter is also not save against overflow. In the final implementation, we should probably give the users access to all available values.

Reply:
Okay.